packages/@webex/internal-plugin-metrics/src/generic-metrics.ts

```diff
@@ -0,0 +1,143 @@
+import {StatelessWebexPlugin} from '@webex/webex-core';
+import {BrowserDetection} from '@webex/common';
+import {merge} from 'lodash';
+import ClientMetricsBatcher from './client-metrics-batcher';
+import {getOSNameInternal} from './metrics';
+import {EventContext, TaggedEvent, EventPayload} from './metrics.types';
+
+const {getOSVersion, getBrowserName, getBrowserVersion} = BrowserDetection();
+
+/**
+ * @description top-level abstract class to handle Metrics and common routines.
+ * @export
+ * @class GenericMetrics
+ */
+export default abstract class GenericMetrics extends StatelessWebexPlugin {
```

Comment:
note: i feel like this could get confusing in the future.. we have this generic metrics class that is used for behavioural, operational and business but not call diagnostic.

Not sure what the best approach could be here..

Reply:
It centralizes the logic of the vast majority of the metric backends (3/4) which doesn't have common concepts, so it feels right to call it generic in this respect. The fact it's not used by one backend doesn't make it less generic or confusing IMO: there is always specific code for specific things, it doesn't make generic things less relevant (as soon as they are mostly used, of course).

That said I'm 100% open to other suggestions, but so far it seems the cleanest approach to me.

Reply:
I don't really have a problem with GenericMetrics. It won't be directly used by anything (consumers will just call the behavioral/business/operational) functions as far as i can tell, so it seems fine to me.

Reply:
Yeah I should probably have clarified this: GenericMetric is really just an internal implementation thing and should never be exposed to the consumer, absolutely.